### PR TITLE
feat(experiments): API to manage web_experiments for no-code experiments

### PR DESCRIPTION
**Pull request by Phanatic**
## Problem

This PR adds a new endpoint that provides the ability to retreive/create/edit and delete web experiments from the Toolbar. 
The `GET` endpoint is used by a new component in `posthog-js` (PR incoming) which will apply the transforms to a page.

#### Changes to data model 
1. Added `type` property to Experiment type, i.e. `posthog_experiment` which can be either product(default) or web.
2. Added `variants` property to Experiment of type JSON which will store information about how the variants transform the web page.

#### Changes to API surface area 
1. Introduces a `GET /projects/@current/web_experiments` which will return data in this format

``` json
{
  "experiments": [
    {
      "id": 15,
      "name": "Zero to Web Experiment",
      "feature_flag_key": "Zero to Web Experiment-feature",
      "variants": {
        "variant #0": {
          "transforms": [
            {
              "html": "",
              "text": "There goes SUperman!",
              "selector": "#page > #body > .header h1"
            }
          ],
          "variant_name": "variant #0",
          "rollout_percentage": 50
        },
        "variant #1": {
          "transforms": [
            {
              "html": "",
              "text": "Up, UP and Away!",
              "selector": "#page > #body > .header h1"
            }
          ],
          "variant_name": "variant #1",
          "rollout_percentage": 50
        }
      }
    }
  ]
}
```

2. Introduces a `PATCH /projects/@current/web_experiments/{id}`  endpoint which will accept data in the same format as shown above and update the web experiment. 

3. Introduces a `DELETE /projects/@current/web_experiments/{id}`  endpoint which will delete the web experiment. 

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

1. Manual testing
- [x] Adding Automated tests now

### CODE REVIEW

File: posthog/api/web_experiment.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                partial=True,
+                context=self.context,
+            )
+            existing_flag_serializer.is_valid(raise_exception=True)
+            existing_flag_serializer.save()
+
+        instance = super().update(instance, validated_data)
+        return instance
+
+    def get_variant_names(self, validated_data: dict[str, Any]):
```
Comment (jurajmajerik) [inline posthog/api/web_experiment.py:131]: This method retrieves variants and converts them to the format used by FeatureFlag. I suggest making this clearer by either renaming it to `get_variants_for_feature_flag` or adding a comment.

**Comment by jurajmajerik** [comment]
Looking good so far :) a few comments

**Comment by dmarticus** [comment]
No additional notes from me!  I'll wait for Juraj's final stamp but things seem reasonable to my understanding.

**Review by dmarticus — APPROVED** [review]
🚢 :shipit:

**Pull request merged**